awscli/s3transfer/crt.py

```diff
@@ -231,6 +232,17 @@ def delete(self, bucket, key, extra_args=None, subscribers=None):
     def shutdown(self, cancel=False):
         self._shutdown(cancel)
 
+    def _validate_checksum_algorithm_supported(self, extra_args):
+        checksum_algorithm = extra_args.get('ChecksumAlgorithm')
+        if checksum_algorithm is None:
+            return
+        if checksum_algorithm not in awscrt.s3.S3ChecksumAlgorithm.__members__:
```

Comment:
Do we guarantee casing will be correct here by calling `upper()` somewhere else? Otherwise, we'd be able to end up with something like `Crc32` right?

Reply:
Nevermind, this isn't configurable by the end user currently so we're only using the constants which should match.

Reply:
I pushed up a commit to make the checksum algorithm case insensitive. In the normal transfer manager, it accepts lowercase values. So it makes sense to accept it here because we will be porting it back to `s3transfer`.